src/libp2p_framed_stream.erl

```diff
@@ -341,6 +493,52 @@ recv(Connection, Timeout) ->
 %% Internal
 %%
 
+-spec verify_exchange(binary(), #state{}) -> {ok, #state{}} | {error, any()}.
+verify_exchange(Data, #state{kind=Kind,
+                             module=Module,
+                             connection=Connection,
+                             send_pid=SendPid,
+                             secured=true,
+                             exchanged=false,
+                             args=Args,
+                             pub_key=PK,
+                             priv_key=SK
+                       }) ->
+    {key_exchange, OtherSidePK, Signature} = erlang:binary_to_term(Data),
```

Comment:
Might want to use the 'safe' option to binary_to_term because this is untrusted input.

Beyond that, it might be better to avoid B2T at all given portability concerns?